### ISSUE TITLE
FCM updated their api.

### ISSUE THREAD

**Issue by Almeonamy** [angelobelchior/FCM.Net#15]
FCM updated their api. Do you plan to update your library for new api version? Now default sounds are not supported in iOS by FCM.Net.
https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages

**Comment by angelobelchior** [comment]
Yes, soon I will make a new version available. Feel free to submit a Pull Request :)

**Comment by nitro52** [comment]
What exactly did they change? based on the docs it looks like Notification only has Title / Body but FCM's notification contains additional properties like Badge. Is this what has changed? Looks like they added platform specific objects at the root as well.

Does this mean badge updates won't work?

**Comment by nitro52** [comment]
Ah i see, FCM.net is using Legacy format. https://firebase.google.com/docs/cloud-messaging/http-server-ref

**Comment by nitro52** [comment]
I started working on this and it seems like a bigger change that i thought. The V1 is an entirely new endpoint that requires the ProjectId for the Url and authenticates using OAuth 2.0.

It's looking like some pretty big change to sender that i was hoping not to do. At the moment i'm just creating a new V1 namespace, that way anyone using the old format should still work. But there seems to be enough changes to question,  should this be a separate package?

**Comment by martinhey** [comment]
In my opinion the legacy implementation should be a separate package, as there are too many changes and I think it would be hard to maintain both api versions in the same library.
However, whatever will be the final decision: I'm currently looking for a good option to send fcm messages using the new API version as I need the separation of values for iOS and Android. So could help to implement the new api version here rather than writing an own implementation. So if you already started implementation, maybe I could contribute.

**Comment by angelobelchior** [comment]
@martinhey I will begin soon to change the api to use the new features. Feel free to contribute :)

**Comment by martinhey** [comment]
of course... let's keep in touch and define subtasks

**Comment by markalanevans** [comment]
@martinhey  so @angelobelchior i'm just starting out, but does this package not support V1? Should I not consider this Package? Perhaps update the Readme.md with the details.

**Comment by ejdrian313** [comment]
is time_to_live change to ttl? because i can't set it in my projects now

**Comment by ofcerci4826** [comment]
@angelobelchior What is the latest situation on this subject?

**Comment by ryburn52** [comment]
@ofcerci4826  We ended up using FCMSharp instead.
